addons/mail/models/mail_channel_partner.py

```diff
@@ -45,3 +45,15 @@ def write(self, vals):
             if {'channel_id', 'partner_id', 'partner_email'} & set(vals):
                 raise AccessError(_('You can not write on this field'))
         return super(ChannelPartner, self).write(vals)
+
+    def mail_channel_partner_format(self, partner_infos=None):
+        members = self.read(['id', 'partner_id'])
+        if partner_infos and len(partner_infos):
+            formatted_members = []
+            for member in members:
+                formatted_members.append({
+                    'id': member['id'],
+                    'partner': partner_infos[member['partner_id'][0]],
+            })
+            return formatted_members
+        return members
```

Comment:
It's confusing to have 2 conditions and 2 different syntax to do the same thing.

I would be expecting to see more clearly the structure of the returned value for each member, as an example similar to what is done in `_message_format` with the final vals update.